pkg/secrets/pushtofile/secret_group.go

```diff
@@ -75,24 +77,33 @@ func (sg *SecretGroup) pushToFileWithDeps(
 		sg.SecretSpecs,
 	)
 	if err != nil {
-		return err
+		return
 	}
 
 	//// Open and push to file
 	wc, err := depOpenWriteCloser(sg.FilePath, sg.FilePermissions)
 	if err != nil {
-		return err
+		return
 	}
+
 	defer func() {
 		_ = wc.Close()
+		if r := recover(); r != nil {
```

Comment:
I think it might aid readability to have a separate defer for the panic recovery for push to write, just before calling push to writer.